packages/payment-processor/test/payment/erc20-transferable-receivable.test.ts

```diff
@@ -0,0 +1,187 @@
+import { Wallet, BigNumber, providers } from 'ethers';
+
+import {
+  ClientTypes,
+  ExtensionTypes,
+  IdentityTypes,
+  RequestLogicTypes,
+} from '@requestnetwork/types';
+import { deepCopy } from '@requestnetwork/utils';
+
+import { approveErc20, getErc20Balance } from '../../src/payment/erc20';
+import {
+  hasReceivableForRequest,
+  mintErc20TransferableReceivable,
+  payErc20TransferableReceivableRequest,
+} from '../../src/payment/erc20-transferable-receivable';
+
+/* eslint-disable no-magic-numbers */
+/* eslint-disable @typescript-eslint/no-unused-expressions */
+
+const erc20ContractAddress = '0x9FBDa871d559710256a2502A2517b794B482Db40';
+
+const mnemonic = 'candy maple cake sugar pudding cream honey rich smooth crumble sweet treat';
+const feeAddress = '0xC5fdf4076b8F3A5357c5E395ab970B5B54098Fef';
+const provider = new providers.JsonRpcProvider('http://localhost:8545');
+const payeeWallet = Wallet.fromMnemonic(mnemonic, "m/44'/60'/0'/0/0").connect(provider);
+const wallet = Wallet.fromMnemonic(mnemonic, "m/44'/60'/0'/0/1").connect(provider);
+const paymentAddress = payeeWallet.address;
+
+const validRequest: ClientTypes.IRequestData = {
+  balance: {
+    balance: '0',
+    events: [],
+  },
+  contentData: {},
+  creator: {
+    type: IdentityTypes.TYPE.ETHEREUM_ADDRESS,
+    value: wallet.address,
+  },
+  currency: 'DAI',
+  currencyInfo: {
+    network: 'private',
+    type: RequestLogicTypes.CURRENCY.ERC20,
+    value: erc20ContractAddress,
+  },
+  events: [],
+  expectedAmount: '100',
+  extensions: {
+    [ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_TRANSFERABLE_RECEIVABLE]: {
+      events: [],
+      id: ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_TRANSFERABLE_RECEIVABLE,
+      type: ExtensionTypes.TYPE.PAYMENT_NETWORK,
+      values: {
+        feeAddress,
+        feeAmount: '0',
+        paymentAddress,
+        salt: '0ee84db293a752c6',
+      },
+      version: '0.1.0',
+    },
+  },
+  payee: {
+    type: IdentityTypes.TYPE.ETHEREUM_ADDRESS,
+    value: paymentAddress,
+  },
+  extensionsData: [],
+  meta: {
+    transactionManagerMeta: {},
+  },
+  pending: null,
+  requestId: '0188791633ff0ec72a7dbdefb886d2db6cccfa98287320839c2f173c7a4e3ce7e1',
+  state: RequestLogicTypes.STATE.CREATED,
+  timestamp: 0,
+  version: '1.0',
+};
+
+describe('erc20-transferable-receivable', () => {
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('mintErc20TransferableReceivable works', () => {
+    it('rejects paying without minting', async () => {
+      const request = deepCopy(validRequest) as ClientTypes.IRequestData;
+
+      const receivableExists = await hasReceivableForRequest(request, wallet);
+
+      if (!receivableExists) {
```

Comment:
Can you remove this if logic and instead assert the expected return from `hasReceivableRequest`? Currently, the test is effectively skipped if `hasReceivableForRequest` mistakenly returns true.

Reply:
Because this test runs on the local chain any subsequent tests after the first one will error when attempting to mint the receivable for a request. Unless the user restarts the chain, this will probably be a confusing case. I'll add a check to make sure the token ID exists, if that works.